common/include/verilog_int.h

```diff
@@ -656,6 +652,45 @@ struct vint {
 		}
 	}
 
+	[[gnu::noinline]]
+	friend void from_bin(vint &val, const ::std::string &s) {
+		constexpr unsigned max_len = num_bit;
+		::std::fill_n(::std::begin(val.v), num_word, 0);
+		if (s.empty()) {
+			return;
+		}
+		unsigned str_pos = s.size()-1;
+		unsigned put_pos = 0;
+		int to_put;
+		// pos == size_t(-1) is safe according to the standard
+		for (put_pos = 0; put_pos < max_len and str_pos != -1; --str_pos) {
+			const char c = s[str_pos];
+			if ('0' <= c and c <= '1') {
+				to_put = c - '0';
+			} else {
+				continue;
+			}
+			val.PutStypeAtBitPosUnsafe(put_pos, to_put);
+			put_pos += 1;
+		}
+		val.ClearUnusedBits();
```

Comment:
Is this line necessary if the content has been cleaned up by `fill_n` in Line 658?

Reply:
No it's not. Unlike hex version, bin version will not set invalid bits in accident.